### PR TITLE
Add option to disable content encoding

### PR DESCRIPTION
**Pull request by AMeng**

### CODE REVIEW

**Comment by AMeng** [comment]
We need the ability to leave off the content-type because Cloud Files automatically determines the content-type based on the file extension.

**Comment by Kami** [comment]
@AMeng I assume you mean Libcloud automatically detects content-type (https://github.com/apache/libcloud/blob/trunk/libcloud/storage/base.py#L597)?

File: file_syncer/run.py [hunk 1]
```diff
@@ -67,6 +67,8 @@ def run():
     parser.add_option('--delete', dest='delete', action='store_true',
                       help='delete extraneous files from dest containers',
                       default=False)
+    parser.add_option('--no-content-type', dest='no_content_type',
```
Comment (Kami) [inline file_syncer/run.py:70]: I would also add `action='store_true', default=False`.

File: file_syncer/run.py [hunk 2]
```diff
@@ -67,6 +67,8 @@ def run():
...
+                      help='do not specify any content type')
```
Comment (Kami) [inline file_syncer/run.py]: I would change the help to something like:

`Don't automatically specify 'application/octet-stream' content-type for every file and use automatic file type detection based on the file extension`

**Comment by Kami** [comment]
Besides couple of things I've mentioned, LGTM

Thanks

**Comment by AMeng** [comment]
Yes, @Kami, that is exactly what I was referring to.

**Comment by Kami** [comment]
Merged, thanks.

**Pull request merged**